app/controllers/orders/items.js

```diff
@@ -22,6 +22,19 @@ export default Ember.Controller.extend(SearchMixin, {
     };
   },
 
+  scannedItem: Ember.observer("searchInput", function() {
+    const searchInput = this.get("searchInput") || "";
+    const sanitizedString = this.sanitizeString(searchInput);
+    if (sanitizedString) {
+      this.set("searchText", sanitizeString);
+    }
+  }),
+
+  scannedText: Ember.observer("searchText", function() {
+    const searchValue = this.get("searchText") || "";
+    this.set("searchInput", this.sanitizeString(searchValue));
```

Comment:
I notice a dangerous loop here, observers can trigger each others in a _seemingly_ infinite loop. Even though it might work for reasons X or Y, that is not a pattern we want to see in the app

Reply:
Observers only trigger if they observe a change in value , so the flow goes as when the searchInput is made set for the first time inside scannedItem then scannedText will be called due to change in search text which at the end is setting searchInput, but this time scannedItem observer wont be triggered and the loop will break because there is no change in the searchInput value. So there is no possibility of any test cases running in infinite loop . @patrixr

Reply:
but there actually is a change to the searchinput : `this.set("searchInput", ...)`

I understand that there's no actual infinite loop, you would have noticed that. It is prevented by X or Y. But writing code that describes that pattern is a big code smell.

Or simply the fact that we have `this.sanitizeString(searchInput)` is weird.

All of the above are indications that there is a simpler and cleaner way to achieve this.